docs/Architecture.md

```diff
@@ -3,6 +3,30 @@ id: architecture
 title: Architecture
 ---
 
+This diagram shows how Jest works to run your tests. Jest is composed of a bunch of internal modules, however, to keep things a bit easier, we've decided to focus at a higher level.
+
+![](https://i.imgur.com/qGVNhBB.png)
```

Comment:
can we host this ourselves instead of on imgur? just adding to https://github.com/facebook/jest/tree/main/website/static/img should be enough

Also, is there an svg version of it?